features/steps/thens/attributes.py

```diff
@@ -61,15 +61,31 @@ def step_impl(context, entity, other_entity):
         misc.handle_errors(context, errors)
 
 
-@then('The type of attribute {attribute} should be {expected_entity_type}')
+@then('The type of attribute {attribute} must be {expected_entity_type}')
 def step_impl(context, attribute, expected_entity_type):
 
     def _():
         for inst in context.instances:
+            if isinstance(inst, (tuple, list)):
+                inst = inst[0]
             related_entity = getattr(inst, attribute, [])
-            if not related_entity.is_a(expected_entity_type):
+            if isinstance(related_entity, (tuple, list)):
+                related_entity = related_entity[0]
+            if isinstance(related_entity, (tuple, list)) or (not related_entity.is_a(expected_entity_type)):
                 yield err.AttributeTypeError(inst, [related_entity], attribute, expected_entity_type)
 
     misc.handle_errors(context, list(_()))
 
 
+@then('The value of attribute {attribute} must be {value}')
+def step_impl(context, attribute, value):
+    if getattr(context, 'applicable', True):
+        errors = []
+        for instance in context.instances:
+            while isinstance(instance, (tuple, list)):
+                instance = instance[0]
+            attribute_value = getattr(instance, attribute, 'Attribute not found')
+            if attribute_value != value:
+                errors.append(err.InvalidValueError(instance, attribute, attribute_value))
+
+        misc.handle_errors(context, errors)
```

Comment:
Good questions IMO.

1. There was no strict convention in Python code, but of course, we tried to keep as consistent as possible. Unified to `inst`. 
2. Used `while `in case there are multiple layers of nesting. Checked a few examples and looks like it might have been an unnecessary safety precaution. Unified to `isinstance`.